chainer/device_resident.py

```diff
@@ -14,6 +15,29 @@
 import chainerx
 
 
+def _warn_to_gpu(dst, arr, legacy):
+    if not legacy:
+        return False
+    if isinstance(dst, cuda.GpuDevice) and isinstance(arr, cuda.ndarray):
+        src_id = arr.device.id
+        dst_id = dst.device.id
+        if src_id == dst_id:
+            # The link is already on the requested device; nothing
+            # to do.
+            return True
+        else:
+            # Sticky option is omitted. As the default behavior is
```

Comment:
This comment is obsolete: there will be no "sticky" option and the behavior won't change. Please fix.